Update-LID using DNN

s-mousmita opened the pull request:
@david-ryan-snyder @danpovey   Adding update.
Changes are made with reference to the old pull request.

david-ryan-snyder commented:
@s-mousmita, thanks for the update.

Could you do the following:
  * Make egs/lre07/v2/lid a symbolic link to egs/lre07/v1/lid instead of a copy
  * Add your new DNN-based LID scripts to egs/lre07/v1/lid

david-ryan-snyder commented:
The point is that we don't want to be maintaining two copies of lid/, one in egs/lre07/v1 and one in egs/lre07/v2 . 

See what we did in egs/sre10/v1/sid and egs/sre10/v2/sid. They both point to egs/sre08/v1/sid .

s-mousmita commented:
Yes, sorry  but script name?

david-ryan-snyder commented on egs/lre07/README.txt:
```diff
@@ -1,9 +1,9 @@
+ Language Evaluation.  The subdirectory v1 demonstrates the standard LID
+ system, which is an I-Vector based recipe using full covarience GMM-UBM and logistic regression model.  
+ The subdirectory v2 demonstrates the LID sysem using a Time Delay Deep Neural Network based UBM 
+ which is used to replace the GMM-UBM of v1. The DNN is trained using about 1800 hours of the english portion of Fisher.
```
Please be mindful of typos here. E.g., covarience -> covariance, and sysem - > system. 

Also, please format the text so that lines don't go over 80 columns.

david-ryan-snyder commented on egs/lre07/v1/lid/extract_ivectors_dnn.sh:
```diff
@@ -0,0 +1,94 @@
+# Copyright     2013  Daniel Povey
+#          2014-2015  David Snyder
+#               2015  Johns Hopkins University (Author: Daniel Garcia-Romero)
+#               2015  Johns Hopkins University (Author: Daniel Povey)
```
Be sure to give yourself credit in the copyright list

david-ryan-snyder commented on egs/lre07/v1/lid/extract_ivectors_dnn.sh:
```diff
@@ -0,0 +1,94 @@
+
+if [ $# != 5 ]; then
+  echo "Usage: $0 <extractor-dir> <data> <ivector-dir>"
+  echo " e.g.: $0 exp/extractor_2048_male data/train_male exp/ivectors_male"
```
This needs to be updated. E.g., script has 5 arguments, but only 3 specified in usage message. And references to male/female don't make sense for LID.

david-ryan-snyder commented on egs/lre07/v1/lid/extract_ivectors_dnn.sh:
```diff
@@ -0,0 +1,94 @@
+sdata_dnn=$data_dnn/split$nj;
+utils/split_data.sh $data_dnn $nj || exit 1;
+
+#delta_opts=`cat $srcdir/delta_opts 2>/dev/null`
```
I don't see any reason to keep the comment

david-ryan-snyder commented on egs/lre07/v1/lid/extract_ivectors_dnn.sh:
```diff
@@ -0,0 +1,94 @@
+
+#delta_opts=`cat $srcdir/delta_opts 2>/dev/null`
+
+splice_opts=`cat exp/nnet//splice_opts 2>/dev/null` # frame-splicing options           
```
I don't think this is used

david-ryan-snyder commented on egs/lre07/v1/lid/extract_ivectors_dnn.sh:
```diff
@@ -0,0 +1,94 @@
+    ivector-normalize-length scp:$dir/ivector.scp  ark:- \| \
+    ivector-mean ark:$data/spk2utt ark:- ark:- ark,t:$dir/num_utts.ark \| \
+    ivector-normalize-length ark:- ark,scp:$dir/spk_ivector.ark,$dir/spk_ivector.scp || exit 1;
+fi
```
Lines 86-94 are SID specific. I don't think we need them for LID

david-ryan-snyder commented on egs/lre07/v1/lid/init_full_ubm_from_dnn.sh:
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+# Copyright 2015   David Snyder
+#           2015   Johns Hopkins University (Author: Daniel Garcia-Romero)
+#           2015   Johns Hopkins University (Author: Daniel Povey)
```
Please credit yourself

david-ryan-snyder commented on egs/lre07/v1/lid/init_full_ubm_from_dnn.sh:
```diff
@@ -0,0 +1,80 @@
+if [ $# != 4 ]; then
+  echo "Usage: steps/init_full_ubm_from_dnn.sh <data-speaker-id> <data-dnn> <dnn-model> <new-ubm-dir>"
+  echo "Initializes a full-covariance UBM from DNN posteriors and speaker recognition features."
+  echo " e.g.: steps/init_full_ubm_from_dnn.sh data/train data/train_dnn exp/dnn/final.mdl exp/full_ubm"
```
Lines 23-25 need to be updated for the LID adaptation of this script

david-ryan-snyder commented on egs/lre07/v1/lid/init_full_ubm_from_dnn.sh:
```diff
@@ -0,0 +1,80 @@
+  echo "  --cmd (utils/run.pl|utils/queue.pl <queue opts>) # how to run jobs."
+  echo "  --nj <n|16>                                      # number of parallel training jobs"
+  echo "  --delta-window <n|3>                             # delta window size"
+  echo "  --delta-order <n|2>                              # delta order"
```
I don't think delta-window or delta-order are used in this script. Please remove.

david-ryan-snyder commented on egs/lre07/v1/lid/train_ivector_extractor_dnn.sh:
```diff
@@ -0,0 +1,180 @@
+# Copyright 2013  Daniel Povey
+#      2014-2015  David Snyder
+#           2015  Johns Hopkins University (Author: Daniel Garcia-Romero)
+#           2015  Johns Hopkins University (Author: Daniel Povey)
```
Please credit yourself

david-ryan-snyder commented on egs/lre07/v1/lid/train_ivector_extractor_dnn.sh:
```diff
@@ -0,0 +1,180 @@
+
+if [ $# != 5 ]; then
+  echo "Usage: $0 <fgmm-model> <dnn-model> <data-speaker-id> <data-dnn> <extractor-dir>"
+  echo " e.g.: $0 exp/sup_ubm/final.ubm exp/dnn/final.mdl data/train data/train_dnn exp/extractor_male"
```
I think this should be updated to remove the reference to extractor_male, since that is SID specific

david-ryan-snyder commented on egs/lre07/v1/lid/train_ivector_extractor_dnn.sh:
```diff
@@ -0,0 +1,180 @@
+utils/split_data.sh $data_dnn $nj_full || exit 1;
+
+
+splice_opts=`cat exp/nnet//splice_opts 2>/dev/null` # frame-splicing options           
```
Is this used in the script?

david-ryan-snyder commented on egs/lre07/v1/lid/train_ivector_extractor_dnn.sh:
```diff
@@ -0,0 +1,180 @@
+  echo $nj_full > $dir/num_jobs
+  echo "$0: doing DNN posterior computation"
+  $cmd JOB=1:$nj_full $dir/log/post.JOB.log \
+  nnet-am-compute --apply-log=true --use-gpu=yes --chunk-size=400 $nnet "$nnet_feats" ark:- \
```
Sorry to reverse what I asked you to do earlier, but I think it will be easier if you change this back to --use-gpu=no. To make GPU support in this script consistent with other Kaldi scripts, you would need to make additional changes that go beyond just setting that one option.

However, you might want to leave a comment in there that says that --use-gpu=yes enables GPU support

david-ryan-snyder commented on egs/lre07/v1/lid/train_ivector_extractor_dnn.sh:
```diff
@@ -0,0 +1,180 @@
+     nt=$num_processes
+    #nt=$[$num_threads*$num_processes] # use the same number of threads that
+                                      # each accumulation process uses, since we
+                                      # can be sure the queue will support this many.
```
I think you should restore the commented out version

david-ryan-snyder commented on egs/lre07/v1/lid/train_ivector_extractor_dnn.sh:
```diff
@@ -0,0 +1,180 @@
+splice_opts=`cat exp/nnet//splice_opts 2>/dev/null` # frame-splicing options           
+
+#parallel_opts="-pe smp $[$num_threads*$num_processes]"
+parallel_opts="-pe smp $num_processes"
```
Probably should restore the commented out version. See comment below about the GPU

david-ryan-snyder commented on egs/lre07/v2/lid:
```diff
@@ -0,0 +1 @@
+/home/mousmita/WORK/LanguageID/kaldi/egs/lre07/v1/lid
```
The symbolic link should be with respect to the example directory, rather than an absolute path in your homedir. E.g., something like ln -s ../v1/lid should do it.

david-ryan-snyder commented on egs/lre07/v2/local/dnn/README:
```diff
@@ -0,0 +1,5 @@
+This directory contains DNN scripts based on the nnet2 recipes found in
+the ASR examples (e.g., fisher_english). The scripts have been modified
+for speaker recognition purposes. Most of the scripts are lightly modified
```
Can you change the wording, to refer to language-id instead of speaker recognition?

david-ryan-snyder commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,244 @@
+local/make_lre03.pl /home/mousmita/DATA/LDC2006S31 data
+local/make_lre05.pl /home/mousmita/DATA/LDC2008S05 data
+local/make_lre07_train.pl /home/mousmita/DATA/LDC2009S05/package/lid07tr1 data
+local/make_lre09.pl /home/mousmita/DATA/LRE2009/eval data
```
Could you change the paths to the ones used in https://github.com/s-mousmita/kaldi/blob/62e6e06da7aaad5e7d34e4aa99fc8f7823a1a3c3/egs/lre07/v1/run.sh ? I think that would be a little more consistent.

david-ryan-snyder commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,244 @@
+# Make the evaluation data set. We're concentrating on the General Language
+# Recognition Closet-Set evaluation, so we remove the dialects and filter
+# out the unknown languages used in the open-set evaluation.
+local/make_lre07.pl /home/mousmita/DATA/LDC2009S04 data/lre07_all
```
Could you also change this path?

david-ryan-snyder commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,244 @@
+#Note:
+# We performed similar experiment using speaker recognition style features (eg. sid/extract_ivectors_dnn.sh) instead of
+# standard SDC features. It can be observed that SDC helps to improve the performance
+# at 10 sec and 30 sec conditions, however at shorter duration like 3 sec speaker recognition style feature is a better choice. 
```
Could you format the comment so that the lines don't extend beyond 80 columns?

david-ryan-snyder commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,244 @@
+			    data/train_lr_dnn \
+			    exp/ivectors_train
+
+lid/extract_ivectors_dnn.sh --cmd "$train_cmd -l mem_free=10G,ram_free=10G" --nj 2 \
```
If you remove the GPU support (see previous comments about it) can you change nj 2 to maybe nj 40?

david-ryan-snyder commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,244 @@
+
+# General LR 2007 closed-set eval
+local/lre07_eval/lre07_eval.sh exp/ivectors_lre07 \
+			         local/general_lr_closed_set_langs.txt
```
Indent on line 222 is a bit strange.

david-ryan-snyder replied:
Would be better to remove one leading space on line 222

david-ryan-snyder commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,244 @@
+
+
+
+
```
Can you remove the extra newlines at the end of the file?

david-ryan-snyder commented:
@s-mousmita, please let me know when you've addressed the comments

s-mousmita commented:
Thanks @david-ryan-snyder .Please check updated scripts.

david-ryan-snyder commented on egs/lre07/v1/lid/init_full_ubm_from_dnn.sh:
```diff
@@ -0,0 +1,80 @@
+
+nnet_feats="ark,s,cs:apply-cmvn-sliding --center=true scp:$sdata_dnn/JOB/feats.scp ark:- |"
+
+## Set up SDC features.
```
Could you remove the extra '#'?

david-ryan-snyder commented on egs/lre07/v1/lid/train_ivector_extractor_dnn.sh:
```diff
@@ -0,0 +1,178 @@
+
+parallel_opts="-pe smp $[$num_threads*$num_processes]"
+
+## Set up features.
```
Could you remove the extra '#'?

david-ryan-snyder commented on egs/lre07/v2/local/callfriend_lang.txt:
```diff
@@ -0,0 +1,7 @@
+55 chinese.mandarin.mainland
+56 chinese.mandarin.taiwan
+57 spanish.caribbean
+58 spanish.noncaribbean
```
I don't see this file referenced anywhere. _If_ it isn't needed, can we remove it?

david-ryan-snyder commented on egs/lre07/v2/local/dnn/run_nnet2_multisplice.sh:
```diff
@@ -0,0 +1,71 @@
+
+if [ $stage -le 6 ]; then
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]]; then
+    utils/create_split_dir.pl /export/b0{6,7,8,9}/$(USER)/kaldi-dsata/egs/fisher_english/s5/$dir/egs/storage $dir/egs/storage
```
Can you change the location the egs are dumped to
`/export/b0{6,7,8,9}/$(USER)/kaldi-data/egs/lre07/v2/$dir/egs/storage`

david-ryan-snyder commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,238 @@
+#utils/fix_data_dir.sh data/train
+#utils/filter_scp.pl data/train/utt2warp data/train/utt2spk > data/train/utt2spk_tmp
+#cp data/train/utt2spk_tmp data/train/utt2spk
+#utils/fix_data_dir.sh data/train
```
Instead of commenting out this block, can we just put it in an if statement? E.g.,
`if $use_vtln; then
  utils/fix_data_dir.sh data/train
  utils/filter_scp.pl data/train/utt2warp data/train/utt2spk > data/train/utt2spk_tmp
  cp data/train/utt2spk_tmp data/train/utt2spk
  utils/fix_data_dir.sh data/train
fi`

david-ryan-snyder replied:
Actually, can we just move that block inside of the previous if statement? After all, it's only entered if use_vtln=true.

david-ryan-snyder commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,238 @@
+# Train an i-vector extractor based on the DNN-UBM.
+lid/train_ivector_extractor_dnn.sh --cmd "$train_cmd -l mem_free=80G,ram_free=80G" \
+				   --min-post 0.015 \
+				   --ivector-dim 600 --nj 1 --num-processes 1 --num-threads 8 \
```
Can you remove the `--nj 1 --num-processes 1 --num-threads 8`? The default options should be OK, and more consistent with similar scripts.

david-ryan-snyder commented:
@s-mousmita, I added a few minor comments. Everything else looks good.

s-mousmita commented:
@david-ryan-snyder, I have updated the scripts accordingly. Thanks.

david-ryan-snyder commented:
@s-mousmita thanks for your work on this, looks great! 

@danpovey unless you have any lingering concerns, would you like to merge it?

danpovey commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,234 @@
+
+# Extract i-vectors using the extractor with the DNN-UBM
+lid/extract_ivectors_dnn.sh --cmd "$train_cmd -l mem_free=10G,ram_free=10G" --nj 40 \
+			    exp/extractor_dnn \
```
we normally use 2 spaces for indent, not 16 or whatever you are using here.  same issue exists elsewhere.

danpovey commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,234 @@
+
+
+# Training data sources
+local/make_sre_2008_train.pl /export/corpora5/LDC/LDC2011S05 data
```
It will make it easier for others to change this for their queues, if you do:
`data_root=/export/corpora/LDC`
and use the variable `$data_root` in place of `/export/corpora*/LDC`.  There are
soft links from /export/corpora to the other numbered locations-- or there should be
(please check that this is true of all these dirs).

david-ryan-snyder replied:
I just verified that all the /export/corpora*/LDC corpora that are used in this example have soft links to /export/corpora/LDC. Using corpora* originates in egs/lre07/v1, which uses the same data. @s-mousmita, when you make this change, would you mind doing the same thing in egs/lre07/v1/run.sh ?

danpovey commented:
When you're done addressing my other comments, please rebase this so there is a single commit.  Search online for "git interactive rebase" to see how.
The issue is that some of your previous commits may have large files, which will bulk up the git repo.  Plus we generally prefer to commit fairly clean histories.

david-ryan-snyder commented:
@s-mousmita, thanks for the update.

@danpovey, as far as I can tell, she addressed those comments. Regarding indentation, there are some NIST perl scripts that have not be reformatted, but I assume those are OK.

s-mousmita commented:
@david-ryan-snyder But have I done the rebase correctly? I think I have done some mistake.

david-ryan-snyder commented:
@s-mousmita I was commenting on the code changes.

I think what @danpovey wants you to do is to _squash_ the previous commits into a single commit. I think this will do it http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git .

david-ryan-snyder commented:
@s-mousmita , thanks for the update. It looks like you successfully squashed all the commits into one commit, and addressed @danpovey  earlier comments. Is it ready to be merged into master?

ngoel17 commented:
I believe so

On Wed, Aug 31, 2016, 5:54 PM david-ryan-snyder <notifications@github.com>
wrote:

> @s-mousmita <https://github.com/s-mousmita> , thanks for the update. It
> looks like you successfully squashed all the commits into one commit, and
> addressed @danpovey <https://github.com/danpovey> earlier comments. Is it
> ready to be merged into master?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-243914683>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AIZyeM6BJ1qpB3z2AU3-CHNXUbfli4UNks5qlfgOgaJpZM4JrEUQ>
> .
>

s-mousmita commented:
@david-ryan-snyder yes, I have addressed all earlier comments of @danpovey and squashed all other commits into one. 
@danpovey sorry for this delay. I am still learning git. 
@david-ryan-snyder thanks for all your help.

danpovey commented:
@david-ryan-snyder, LMK if it's good to check in.

david-ryan-snyder commented on egs/lre07/README.txt:
```diff
@@ -21,7 +23,7 @@
    CALLFRIEND Caribbean Spanish:         LDC96S57
    CALLFRIEND Non-Caribbean Spanish:     LDC96S58
-   LRE 1996:                             LDC?
+   LRE 1996:                             lid96e1
```
LRE 1996 is actually included with LDC2006S31, so you can replace lid96e1 with LDC2006S31.

david-ryan-snyder commented on egs/lre07/v1/run.sh:
```diff
@@ -14,31 +14,32 @@ mfccdir=`pwd`/mfcc
+local/make_callfriend.pl $data_root/LDC96S56 chinese.mandarin.taiwan data
+local/make_callfriend.pl $data_root/LDC96S57 spanish.caribbean data
+local/make_callfriend.pl $data_root/LDC96S58 spanish.noncaribbean data
 local/make_lre96.pl /export/corpora/NIST/lid96e1 data
```
It turns out that preparing LRE 1996 is redundant, since it is included with LDC2006S31 which is prepared next. Can you remove this line? The results will _not_ be affected since the data/train_unsplit is the same either way.

david-ryan-snyder commented on egs/lre07/v1/lid/extract_ivectors_dnn.sh:
```diff
@@ -0,0 +1,83 @@
+if [ $stage -le 0 ]; then
+  echo "$0: extracting iVectors"
+  $cmd JOB=1:$nj $dir/log/extract_ivectors.JOB.log \
+    nnet-am-compute --apply-log=true --use-gpu=yes --chunk-size=100 $nnet "$nnet_feats" ark:- \
```
Could you remove the use-gpu=yes and chunk-size=100? Enabling GPU support in this script and making it consistent in other scripts is going to require more than setting that option. It's better to make it default to CPU for now. This can get updated later when we add GPU support to the similar scripts in egs/sre10/v2.

david-ryan-snyder commented on egs/lre07/v1/lid/init_full_ubm_from_dnn.sh:
```diff
@@ -0,0 +1,80 @@
+num_components=`grep -oP 'output-dim\ \K[0-9]+' <(nnet-am-info $nnet 2> /dev/null)`
+
+$cmd JOB=1:$nj $logdir/make_stats.JOB.log \
+  nnet-am-compute --apply-log=true --use-gpu=yes --chunk-size=100 $nnet "$nnet_feats" ark:- \
```
Please remove the use-gpu=yes and chunk-size=100, since the script isn't really set up properly to use the GPU.

david-ryan-snyder commented on egs/lre07/v1/lid/train_ivector_extractor_dnn.sh:
```diff
@@ -0,0 +1,178 @@
+
+
+if [ $# != 5 ]; then
+  echo "Usage: $0 <fgmm-model> <dnn-model> <data-speaker-id> <data-dnn> <extractor-dir>"
```
The usage message should should say "data-language-id", not "data-speaker-id."

david-ryan-snyder commented on egs/lre07/v2/local/make_lre96.pl:
```diff
@@ -0,0 +1,80 @@
+  system("utils/fix_data_dir.sh $out_dir");
+  (system("utils/validate_data_dir.sh --no-text --no-feats $out_dir") == 0) 
+    || die "Error validating data dir.";
+}
```
You can remove this file. It is made redundant by LDC2006S31.

david-ryan-snyder commented on egs/lre07/v2/run.sh:
```diff
@@ -0,0 +1,234 @@
+local/make_callfriend.pl $data_root/LDC96S56 chinese.mandarin.taiwan data
+local/make_callfriend.pl $data_root/LDC96S57 spanish.caribbean data
+local/make_callfriend.pl $data_root/LDC96S58 spanish.noncaribbean data
+local/make_lre96.pl /export/corpora/NIST/lid96e1 data
```
You can remove line 39, since all the data in lid96e1 is included in LDC2006S31.

david-ryan-snyder commented on egs/lre07/v1/lid/extract_ivectors_dnn.sh:
```diff
@@ -0,0 +1,83 @@
+
+
+if [ $# != 5 ]; then
+  echo "Usage: $0  <extractor-dir> <dnn-model> <data-langauge-id> <data-dnn> <ivectors-dir>"
```
langauge -> language

s-mousmita commented:
@david-ryan-snyder Thanks.  Corrected.

david-ryan-snyder commented:
@danpovey, I think it's ready to check in.

danpovey commented:
I just had a look at it...
I think it would be better to move the scripts that are currently in local/dnn/ -- at least the scripts that were copied from steps/nnet2-- to be moved to lid/nnet2/.  That way, those scripts won't have to be duplicated if someone wants to create another setup using the same methods.

Also be careful that sometimes the program pathname appears in usage messages (it's wrong in some cases currently)-- grepping for steps/nnet2 and local/dnn in those scripts should find all the places you'd need to change the path.

s-mousmita commented:
I have removed local/dnn and created lid/nnet2. Paths local/dnn and steps/nnet2 at various places of the scripts have been changed to lid/nnet2.

david-ryan-snyder commented:
I don't think we should move everything from local/dnn to lid/nnet2. My suggestion:

* Recipe specific things go into local. This includes scripts that refer to datasets, or specific instantiations of nnet configurations. That means that train_dnn.sh, run_nnet2_common.sh, run_nnet2_multisplice.sh, Fisher data preparation scripts, etc, should remain in local/dnn
* General purpose scripts can be used by any LID example should go into lid. Scripts such as get_egs2.sh, get_lda.sh, make_multisplice_configs.py, relabel_egs2.sh, etc would go into lid/nnet2
* See the contents of egs/fisher_english/s5/local for examples of what should be in local, if in doubt
* Also, since we're refactoring things, please double check that all the scripts you're moving around are actually needed. For example, I think we don't need an lid/nnet2/remove_egs.sh. There's probably an identical one in steps/nnet2 that can be called instead.

s-mousmita commented:
@david-ryan-snyder  The following are correct? 
- get_egs2.sh, relabel_egs2.sh and get_lda.sh will go to /lid/nnet2/
- train_dnn.sh, fisher_data_prep.sh, fisher_prepare_dict.sh, fisher_train_lms.sh, fisher_create_test_lang.sh,  remove_dup_utts.sh, run_nnet2_multisplice.sh, run_nnet2_common.sh and train_multisplice_accel2.sh will go to local/dnn/
- get_num_frames.sh, make_multisplice_configs.py, align.sh and remove_egs.sh will be used from steps/nnet2.

danpovey commented:
train_multisplice_accel2.sh  is not dataset-specific so it should go into
lid/nnet2.  Otherwise that looks plausible.

Dan


On Sat, Sep 3, 2016 at 1:47 PM, Mousmita Sarma <notifications@github.com>
wrote:

> @david-ryan-snyder <https://github.com/david-ryan-snyder> The following
> are correct?
>
>    - get_egs2.sh, relabel_egs2.sh and get_lda.sh will go to /lid/nnet2/
>    - train_dnn.sh, fisher_data_prep.sh, fisher_prepare_dict.sh,
>    fisher_train_lms.sh, fisher_create_test_lang.sh, remove_dup_utts.sh,
>    run_nnet2_multisplice.sh, run_nnet2_common.sh and
>    train_multisplice_accel2.sh will go to local/dnn/
>    - get_num_frames.sh, make_multisplice_configs.py, align.sh and
>    remove_egs.sh will be used from steps/nnet2.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-244560182>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu66_Rg_P4Tau5tSHXIZI8HVpi4X_ks5qmbKegaJpZM4JrEUQ>
> .
>

s-mousmita commented:
I have made the changes suggested above.

danpovey commented:
Would you mind trying to run it one more time before we commit it?
With all the name changes, there is a strong possibility of errors.


On Sat, Sep 3, 2016 at 3:41 PM, Mousmita Sarma <notifications@github.com>
wrote:

> I have made the changes suggested above.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-244566010>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu1Qg_Jq5SHalGd05kD10su6i-x9uks5qmc1QgaJpZM4JrEUQ>
> .
>

david-ryan-snyder commented:
I agree with the script locations in the current commit.

ngoel17 commented:
Dan,
   Would it make sense to run it on JHU machines so that we don't change
even the dataset paths?

Nagendra

On Sat, Sep 3, 2016, 3:42 PM Daniel Povey <notifications@github.com> wrote:

> Would you mind trying to run it one more time before we commit it?
> With all the name changes, there is a strong possibility of errors.
>
>
> On Sat, Sep 3, 2016 at 3:41 PM, Mousmita Sarma <notifications@github.com>
> wrote:
>
> > I have made the changes suggested above.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-244566010>,
> or mute
> > the thread
> > <
> https://github.com/notifications/unsubscribe-auth/ADJVu1Qg_Jq5SHalGd05kD10su6i-x9uks5qmc1QgaJpZM4JrEUQ
> >
> > .
> >
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-244566067>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AIZyeB_vZIPgipJu7Tsj1ZYNF43NCbcnks5qmc2ggaJpZM4JrEUQ>
> .
>

danpovey commented:
Yes, that sounds good.

On Sun, Sep 4, 2016 at 12:24 AM, Nagendra Goel <notifications@github.com>
wrote:

> Dan,
> Would it make sense to run it on JHU machines so that we don't change
> even the dataset paths?
>
> Nagendra
>
> On Sat, Sep 3, 2016, 3:42 PM Daniel Povey <notifications@github.com>
> wrote:
>
> > Would you mind trying to run it one more time before we commit it?
> > With all the name changes, there is a strong possibility of errors.
> >
> >
> > On Sat, Sep 3, 2016 at 3:41 PM, Mousmita Sarma <notifications@github.com
> >
> > wrote:
> >
> > > I have made the changes suggested above.
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-244566010>,
> > or mute
> > > the thread
> > > <
> > https://github.com/notifications/unsubscribe-auth/ADJVu1Qg_
> Jq5SHalGd05kD10su6i-x9uks5qmc1QgaJpZM4JrEUQ
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you commented.
> >
> >
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-244566067>,
> or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/AIZyeB_
> vZIPgipJu7Tsj1ZYNF43NCbcnks5qmc2ggaJpZM4JrEUQ>
> > .
>
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-244583638>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu4WFbD8HHG2d5WteVcc_01amH29Cks5qmkgWgaJpZM4JrEUQ>
> .
>

danpovey commented:
I assume someone is still working on this?

ngoel17 commented:
Training is running from scratch on JHU machines.

On Tue, Sep 20, 2016, 7:41 PM Daniel Povey <notifications@github.com> wrote:

> I assume someone is still working on this?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-248468766>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AIZyeMZ4AW1pW_COTsTbZ4qYxVqSf3lXks5qsG83gaJpZM4JrEUQ>
> .
>

danpovey commented:
@ngoel17, any progress on this?

s-mousmita commented:
@danpovey @ngoel17  Its doing the last steps, I-vector extraction.

s-mousmita commented:
I have updated the scripts.

david-ryan-snyder commented on egs/lre07/v1/lid/nnet2/train_multisplice_accel2.sh:
```diff
@@ -0,0 +1,641 @@
+        --transform-dir "$transform_dir" --online-ivector-dir "$online_ivector_dir" \
+        --iter $x $data $lang $dir $dir/ali_$time || exit 1
+
+      lid/nnet2/relabel_egs2.sh --cmd "$cmd" --iter $x $dir/ali_$time \
```
Are you sure we can't use steps/nnet2/relabel_egs2.sh here?

david-ryan-snyder commented on egs/lre07/v1/lid/nnet2/relabel_egs2.sh:
```diff
@@ -0,0 +1,90 @@
+    $egs_in $egs_out || exit 1
+fi
+
+echo "$0: Finished relabeling training examples"
```
Are you sure we need this script? Can we not use the one in steps/nnet2/relabel_egs2.sh?

david-ryan-snyder commented on egs/lre07/v2/cmd.sh:
```diff
@@ -0,0 +1,28 @@
+export decode_cmd=run.pl
+export cuda_cmd=run.pl
+export mkgraph_cmd=run.pl
+
```
The way the queue options are handled was changed in this commit: https://github.com/kaldi-asr/kaldi/commit/b3bbc038681b5bcd6be02481785f929b5b99e11a

Basically you need to update the comment and the export statements to be more similar to that? The default commands should include queue.pl in it.

david-ryan-snyder commented on egs/lre07/v2/local/lre07_eval/score_lre07.v01d.pl:
```diff
@@ -0,0 +1,1137 @@
+	    return -1;
+	}
+    }
+}
```
For files like this, that are unlikely to ever be modified, it's better to make a symbolic link to the same file in egs/lre07/v1/local.

david-ryan-snyder replied:
You probably don't need to do that for scripts that we wrote that are likely to be modified by the user.

david-ryan-snyder commented on egs/lre07/README.txt:
```diff
@@ -1,9 +1,11 @@
+ Language Evaluation.  The subdirectory v1 demonstrates the standard
+ LID system, which is an I-Vector based recipe using full covariance
+ GMM-UBM and logistic regression model. The subdirectory v2 demonstrates
+ the LID system using a Time Delay Deep Neural Network based UBM
```
I would consider replacing "Time Delay Deep Neural Network" with "time-delay deep neural network."

s-mousmita commented:
@david-ryan-snyder Scripts are updated according to your recent comments.

david-ryan-snyder commented:
@danpovey it looks like the error is in exp-test. Have you seen this in recent PRs?

`Successfully configured OpenBLAS from /home/travis/xroot/usr.
/usr/bin/ld: cannot find -lgfortran
collect2: ld returned 1 exit status
make: *** [exp-test] Error 1
./configure: line 222: ./exp-test: No such file or directory`

danpovey commented:
That exp-test thing was not what was causing the build to fail, it's normal
on the travis build.
The reason was:
  Your test run exceeded 120 minutes.
Probably just random- there were no source changes in the PR.
I'll probably merge today.


On Sun, Oct 30, 2016 at 11:41 AM, david-ryan-snyder <
notifications@github.com> wrote:

> @danpovey <https://github.com/danpovey> it looks like the error is in
> exp-test. Have you seen this in recent PRs?
>
> Successfully configured OpenBLAS from /home/travis/xroot/usr.
> /usr/bin/ld: cannot find -lgfortran
> collect2: ld returned 1 exit status
> make: *** [exp-test] Error 1
> ./configure: line 222: ./exp-test: No such file or directory
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/999#issuecomment-257158438>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu1s9bmy0cJqaVEfRTKoZsXBuR2stks5q5LqhgaJpZM4JrEUQ>
> .
>

danpovey commented:
Thanks!  Merging.

s-mousmita commented:
@danpovey @david-ryan-snyder Thank you for approving the recipe.
